From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Robin Snyders <robin@snyders.xyz>
Date: Wed, 5 Aug 2026 21:09:59 +0200
Subject: [PATCH] ASoC: sm8150: drive Hotdog speaker clocks

---
 sound/soc/qcom/sm8150.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/qcom/sm8150.c b/sound/soc/qcom/sm8150.c
index 7451713b86fa..5b31489e9202 100644
--- a/sound/soc/qcom/sm8150.c
+++ b/sound/soc/qcom/sm8150.c
@@ -317,7 +317,8 @@ static int sm8150_snd_startup(struct snd_pcm_substream *substream)
 		snd_soc_dai_set_sysclk(cpu_dai,
 			Q6AFE_LPASS_CLK_ID_QUAD_MI2S_IBIT,
 			MI2S_BCLK_RATE, SNDRV_PCM_STREAM_PLAYBACK);
-		snd_soc_dai_set_fmt(cpu_dai, fmt);
+		/* The external speaker amplifiers consume BCLK and frame sync. */
+		snd_soc_dai_set_fmt(cpu_dai, SND_SOC_DAIFMT_BP_FP);
 		break;
 	case QUATERNARY_TDM_RX_0:
 		codec_dai_fmt |= SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_DSP_A;
