From 85c6a4ee2f9fc59e8924aa87505a1a796a22fc18 Mon Sep 17 00:00:00 2001
From: Robin Snyders <robin@snyders.xyz>
Date: Sun, 16 Aug 2026 00:33:05 +0200
Subject: [PATCH] usb: gadget: allow 900 mA SuperSpeed VBUS draw

Signed-off-by: Robin Snyders <robin@snyders.xyz>
---
 drivers/usb/gadget/Kconfig | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/gadget/Kconfig b/drivers/usb/gadget/Kconfig
index 76521555e3c1..3818e5942bf1 100644
--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -97,8 +97,8 @@ config USB_GADGET_DEBUG_FS
 	   to conserve kernel memory, say "N".
 
 config USB_GADGET_VBUS_DRAW
-	int "Maximum VBUS Power usage (2-500 mA)"
-	range 2 500
+	int "Maximum VBUS Power usage (2-900 mA)"
+	range 2 900
 	default 2
 	help
 	   Some devices need to draw power from USB when they are
@@ -107,8 +107,9 @@ config USB_GADGET_VBUS_DRAW
 	   such as an AC adapter or batteries.
 
 	   Enter the maximum power your device draws through USB, in
-	   milliAmperes.  The permitted range of values is 2 - 500 mA;
-	   0 mA would be legal, but can make some hosts misbehave.
+	   milliAmperes.  The permitted range of values is 2 - 900 mA;
+	   USB 2.x configurations are capped at 500 mA.  0 mA would be
+	   legal, but can make some hosts misbehave.
 
 	   This value will be used except for system-specific gadget
 	   drivers that have more specific information.
